pkg/controllers/status/cluster_status_controller.go

```diff
@@ -254,9 +264,9 @@ func (c *ClusterStatusController) buildInformerForCluster(cluster *clusterv1alph
 	// create the informer for pods and nodes
 	allSynced := true
 	for _, gvr := range gvrs {
-		if !singleClusterInformerManager.IsInformerSynced(gvr) {
+		if !singleClusterInformerManager.IsInformerSynced(gvr) || !singleClusterInformerManager.IsHandlerExist(gvr, c.getEventHandler(cluster.Name)) {
```

Comment:
Informer synced for `gvr` doesn't mean the `eventHandler` exit. 
Since all controllers share the `MultiClusterInformerManager`, the informer might be created(and synced from another place).